Wikipedia/UI-V5/WMFArticleListCollectionViewController.m

```diff
@@ -198,43 +219,54 @@ - (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id <UIV
     } completion:NULL];
 }
 
-#pragma mark - UICollectionViewDelegate
+#pragma mark - Article Selection
 
-- (void)collectionView:(UICollectionView*)collectionView didSelectItemAtIndexPath:(NSIndexPath*)indexPath {
-    self.selectedArticle = [self.dataSource articleForIndexPath:indexPath];
+- (void)showArticle:(MWKArticle*)article discoveryMethod:(MWKHistoryDiscoveryMethod)discoveryMethod {
+    if (self.delegate) {
+        [self.delegate didSelectArticle:article sender:self];
+        return;
+    }
 
     WMFArticleContainerViewController* container =
```

Comment:
That's the part of this I don't like. I'd really like to DRY that up, but I'm not sure of the best way to do it.  We can brainstorm of hangout if you want.  The main issue is the who/where article container view controllers are configured & presented. So far the options I've come up with are:

- Use the `WMFSearchPresentationDelegate` accessors (which is a superset of those needed to create an article container) to create the article container, then present it
- Call `showViewController:sender:` and let the presenter handle configuration & presentation
- implement lazy getters for `WMFArticleContainerViewController` dependencies which grab the "main" instances of what it needs (data store, saved page list, etc.)